### PR TITLE
Fixed #30271 -- Added the Sign database function.

### PR DESCRIPTION
**Pull request by pope1ni**
Ticket [#30271](https://code.djangoproject.com/ticket/30271).

### CODE REVIEW

**Comment by pope1ni** [comment]
buildbot, test on oracle.

**Comment by pope1ni** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by pope1ni** [comment]
buildbot, test on oracle.

**Comment by pope1ni** [comment]
buildbot, test on oracle.

**Comment by pope1ni** [comment]
buildbot, test on oracle.

**Comment by pope1ni** [comment]
(Added the test module I previously forgot to commit.)

**Comment by pope1ni** [comment]
Test failure is unrelated.

File: docs/ref/models/database-functions.txt [hunk 1]
```diff
@@ -1099,6 +1099,31 @@ It can also be registered as a transform. For example::
...
+    >>> vector = Vector.objects.annotate(x_sign=Sign('x'), y_sign=Sign('y')).get()
+    >>> vector.x_sign, vector.y_sign
+    (1, -1)
+
+It can also be registered as a transform. For example::
+
+    >>> from django.db.models import FloatField
+    >>> from django.db.models.functions import Sign
+    >>> FloatField.register_lookup(Sign)
+    >>> # Get vectors whose sign() is less than 0
```
Comment (felixxm) [inline docs/ref/models/database-functions.txt]: `# Get vectors whose sign() is less than 0` -> 
`# Get vectors whose signs of components are less than 0.`

File: tests/db_functions/math/test_sign.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self.assertIsInstance(obj.big_sign, int)
+        self.assertAlmostEqual(obj.small_sign, -1)
+        self.assertAlmostEqual(obj.normal_sign, 0)
+        self.assertAlmostEqual(obj.big_sign, 1)
+
+    def test_transform(self):
+        with register_lookup(DecimalField, Sign):
+            DecimalModel.objects.create(n1=Decimal('5.4'), n2=Decimal('0'))
+            DecimalModel.objects.create(n1=Decimal('-0.1'), n2=Decimal('0'))
+            obj = DecimalModel.objects.filter(n1__sign__lt=0).get()
```
Comment (felixxm) [inline tests/db_functions/math/test_sign.py]: Maybe we can add also `, n2__sign=0`.

File: tests/db_functions/math/test_sign.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+from django.db.models import DecimalField
+from django.db.models.functions import Sign
+from django.test import TestCase
+from django.test.utils import register_lookup
+
+from ..models import DecimalModel, FloatModel, IntegerModel
+
+
+class SignTests(TestCase):
+
```
[the remark this answers is not in the document]
Comment (pope1ni) [reply tests/db_functions/math/test_sign.py]: That would be inconsistent with all of the other test modules in `tests/db_functions/math/`.

File: tests/db_functions/math/test_sign.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        obj = IntegerModel.objects.annotate(null_sign=Sign('normal')).first()
+        self.assertIsNone(obj.null_sign)
+
+    def test_decimal(self):
+        DecimalModel.objects.create(n1=Decimal('-12.9'), n2=Decimal('0.6'))
+        obj = DecimalModel.objects.annotate(n1_sign=Sign('n1'), n2_sign=Sign('n2')).first()
+        self.assertIsInstance(obj.n1_sign, Decimal)
+        self.assertIsInstance(obj.n2_sign, Decimal)
+        self.assertAlmostEqual(obj.n1_sign, Decimal('-1'))
+        self.assertAlmostEqual(obj.n2_sign, Decimal('1'))
```
[the remark this answers is not in the document]
Comment (pope1ni) [reply tests/db_functions/math/test_sign.py]: Agreed. Makes sense as there are only three possible values.

**Comment by pope1ni** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Pull request merged**